lms/services/vitalsource.py

```diff
@@ -0,0 +1,60 @@
+import oauthlib
+from oauthlib.oauth1 import SIGNATURE_HMAC_SHA1, SIGNATURE_TYPE_BODY
+
+
+class VitalSourceService:
+    def __init__(self, _context, request):
+        self._oauth_key = request.registry.settings["vitalsource_launch_key"]
+        self._oauth_secret = request.registry.settings["vitalsource_launch_secret"]
+
+    def get_launch_params(self, book_id, cfi, lti_user):
+        """
+        Launch the VitalSource book viewer using an LTI launch.
+
+        See https://developer.vitalsource.com/hc/en-us/articles/215612237-POST-LTI-Create-a-Bookshelf-Launch
+        """
+
+        launch_url = f"https://bc.vitalsource.com/books/{book_id}"
```

Comment:
The book ID can be included either in the launch URL or in launch params via the `custom_vbid` field. I went with using the launch URL because that seemed the most obvious thing to do. Putting in the params in future could have the potential advantage of allowing the launch URL to be removed from the Python code and just hardcoded in the LMS frontend.